biotrainer/config/configurator.py

```diff
@@ -317,7 +392,273 @@ def _get_config_maps(
+            )
+
+        sequences = []
+        targets = []
+        set_values = []
+
+        if len(available_splits) == 3:
+            logger.info(f"Three splits found: {available_splits}. Processing each split separately.")
+            for split_name in available_splits:
+                split_dataset = dataset[split_name]
+                if split_dataset is None:
+                    logger.warning(f"Split '{split_name}' is None. Skipping.")
+                    continue
+
+                # Verify that the required columns exist
+                self._verify_columns(split_dataset, sequence_column, target_column)
+
+                # Determine SET value based on split name
+                set_name = self._determine_set_name(split_name)
+
+                split_sequences = split_dataset[sequence_column]
+                split_targets = split_dataset[target_column]
+
+                sequences.extend(split_sequences)
+                targets.extend(split_targets)
+                set_values.extend([set_name] * len(split_sequences))
+
+        else:
+            # If there are not exactly 3 splits, merge all available splits into one
+            logger.info(f"Merging {len(available_splits)} splits and performing a train/val/test split.")
+
+            combined_sequences = []
+            combined_targets = []
+
+            # Loop through all available splits and merge them
+            for split_name in available_splits:
+                split_dataset = dataset[split_name]
+                if split_dataset is None:
+                    logger.warning(f"Split '{split_name}' is None. Skipping.")
+                    continue
+
+                # Verify that the required columns exist for each split
+                self._verify_columns(split_dataset, sequence_column, target_column)
+
+                split_sequences = split_dataset[sequence_column]
+                split_targets = split_dataset[target_column]
+
+                combined_sequences.extend(split_sequences)
+                combined_targets.extend(split_targets)
+
+            # Perform the split: 70% train, 15% val, 15% test
+            train_seq, temp_seq, train_tgt, temp_tgt = train_test_split(
+                combined_sequences, combined_targets, test_size=0.30, random_state=42, shuffle=True
+            )
+            val_seq, test_seq, val_tgt, test_tgt = train_test_split(
+                temp_seq, temp_tgt, test_size=0.50, random_state=42, shuffle=True
+            )
+
+            # Assign SET values
+            sequences.extend(train_seq)
+            targets.extend(train_tgt)
+            set_values.extend(["TRAIN"] * len(train_seq))
+
+            sequences.extend(val_seq)
+            targets.extend(val_tgt)
+            set_values.extend(["VAL"] * len(val_seq))
+
+            sequences.extend(test_seq)
+            targets.extend(test_tgt)
+            set_values.extend(["TEST"] * len(test_seq))
```

Comment:
This is in general correct and nice, but I'd rather not implement that at this point. The thing is that until now we always expect the user come up with pre-defined splits in the input files. Doing dataset splitting on the go would be a separate feature for me which can be done generally for any input data, later in the pipeline.

Reply:
I checked some datasets and noticed that a significant number of Hugging Face datasets only have a single train split. Some of them also have two splits, such as train and test, which we can't handle now (since we need train, val, test). So, I thought it would be a good idea to address those cases as well. But if we're going to handle them later in the pipeline, it is even better.